### PR TITLE
Use version for action azure/webapps-deploy instead of hash

### LINKED ISSUE #25780: Command in guide Deploying Docker to Azure App Service no longer working
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

[Deploying Docker to Azure App Service](https://docs.github.com/en/actions/deployment/deploying-to-your-cloud-provider/deploying-to-azure/deploying-docker-to-azure-app-service)

### What part(s) of the article would you like to see updated?

When running the action in the section [Creating the workflow](https://docs.github.com/en/actions/deployment/deploying-to-your-cloud-provider/deploying-to-azure/deploying-docker-to-azure-app-service#creating-the-workflow) there is an issue with the Step "Deploy to Azure Web App".

When running the follwing error message is returned:

```
Run azure/webapps-deploy@05ac4e98bfa0f856e6669624239291c73ca27698
node:internal/modules/cjs/loader:936
  throw err;
  ^

Error: Cannot find module '@actions/core'
Require stack:
- /home/runner/work/_actions/azure/webapps-deploy/05ac4e98bfa0f856e6669624239291c73ca27698/lib/main.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/home/runner/work/_actions/azure/webapps-deploy/05ac4e98bfa0f856e6669624239291c73ca27698/lib/main.js:19:27)
    at Module._compile (node:internal/modules/cjs/loader:1105:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/home/runner/work/_actions/azure/webapps-deploy/05ac4e98bfa0f856e6669624239291c73ca27698/lib/main.js'
  ]
}
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by samuel-coderhead**
### Why:

When changing to version instead of specific commit the action runs as expected.

Closes: https://github.com/github/docs/issues/25780

### What's being changed (if available, include any code snippets, screenshots, or gifs):
Remove specific commit and use version instead.

### Check off the following:

- [X] I have reviewed my changes in staging, available via the **View deployment** link in this PR's timeline.

  - For content changes, you will also see an automatically generated comment with links directly to pages you've modified. The comment won't appear if your PR only edits files in the `data` directory.
- [X] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### CODE REVIEW

**Comment by cmwilson21** [comment]
@samuel-coderhead Thanks for submitting a PR! I've edited your post slightly to [connect it to your issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue). 

I'll get this triaged for review! :zap:

File: content/actions/deployment/deploying-to-your-cloud-provider/deploying-to-azure/deploying-nodejs-to-azure-app-service.md [hunk 1]
```diff
@@ -121,7 +121,7 @@ jobs:
 
     - name: 'Deploy to Azure WebApp'
       id: deploy-to-webapp 
-      uses: azure/webapps-deploy@05ac4e98bfa0f856e6669624239291c73ca27698
+      uses: azure/webapps-deploy@v2
```
Comment (skedwards88) [inline content/actions/deployment/deploying-to-your-cloud-provider/deploying-to-azure/deploying-nodejs-to-azure-app-service.md:124]:
```suggestion
      uses: azure/webapps-deploy@85270a1854658d167ab239bce43949edb336fa7c
```

File: content/actions/deployment/deploying-to-your-cloud-provider/deploying-to-azure/deploying-php-to-azure-app-service.md [hunk 2]
```diff
@@ -141,7 +141,7 @@ jobs:
 
       - name: 'Deploy to Azure Web App'
         id: deploy-to-webapp
-        uses: azure/webapps-deploy@05ac4e98bfa0f856e6669624239291c73ca27698
+        uses: azure/webapps-deploy@v2
```
Comment (skedwards88) [inline content/actions/deployment/deploying-to-your-cloud-provider/deploying-to-azure/deploying-php-to-azure-app-service.md:144]:
```suggestion
        uses: azure/webapps-deploy@85270a1854658d167ab239bce43949edb336fa7c
```

File: content/actions/deployment/deploying-to-your-cloud-provider/deploying-to-azure/deploying-python-to-azure-app-service.md [hunk 3]
```diff
@@ -134,7 +134,7 @@ jobs:
 
       - name: 'Deploy to Azure Web App'
         id: deploy-to-webapp
-        uses: azure/webapps-deploy@05ac4e98bfa0f856e6669624239291c73ca27698
+        uses: azure/webapps-deploy@v2
```
Comment (skedwards88) [inline content/actions/deployment/deploying-to-your-cloud-provider/deploying-to-azure/deploying-python-to-azure-app-service.md:137]:
```suggestion
        uses: azure/webapps-deploy@85270a1854658d167ab239bce43949edb336fa7c
```

**Review by skedwards88 — COMMENTED** [review]
Thanks for this fix! We require third party actions to use a SHA (https://github.com/github/docs-internal/blob/main/contributing/content-style-guide.md#pinning-version-numbers-to-sha), so I've updated your changes to use the SHA of the latest release (v2.2.9) instead. I'll incorporate those suggestions, then we'll get this merged for you.

**Review by skedwards88 — APPROVED** [review]

**Pull request merged**